internal/broker/broker.go

```diff
@@ -550,13 +550,30 @@ func (b *Broker) CancelIsAuthenticated(sessionID string) {
 }
 
 // UserPreCheck checks if the user is valid and can be allowed to authenticate.
-func (b *Broker) UserPreCheck(username string) error {
+func (b *Broker) UserPreCheck(username string) (string, error) {
+	found := false
 	for _, suffix := range b.allowedSSHSuffixes {
 		if strings.HasSuffix(username, suffix) {
-			return nil
+			found = true
+			break
 		}
 	}
-	return errors.New("username does not match the allowed suffixes")
+
+	if !found {
+		return "", errors.New("username does not match the allowed suffixes")
+	}
+
+	u := userInfo{
```

Comment:
I would factor userInfo creation (so that if one day, we give the option the option to change the shell or expand other options with sane defaults, this is reflected there), wdyt?

Reply:
Sounds like a good idea, for sure. I'll update it.

Reply:
I’m unsure we understood each other, what I meant here is that the 2 paths we are creating userInfo should be the same function call which is something like newUserInfo(). Does it makes sense?

Reply:
I'm actually ashamed that my brain turned so off for that change :joy: